### PR TITLE
Add PowerScale v2.4.0 support

### PR DESCRIPTION
**Pull request by jooseppi-luna**
# Description
Adding PowerScale v2.4.0 support to csm-operator.
- Create files for v2.4.0 support
- Update extensions and sidecar versions
- Change authorization `INSECURE` flag to `SKIP_CERTIFICATE_VALIDATION` in v1.4.0

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/350 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Installed PowerScale 2.4.0 with new files, confirmed working. Used `dellemc/csi-isilon:nightly` since v2.4.0 is not released yet.

### CODE REVIEW

**Comment by jooseppi-luna** [comment]
*Please note:* The image scanner failure is not due to the changes in this PR, as I also ran actions on a fresh branch that I cut off of main and made no changes to and got the same errors:

https://github.com/dell/csm-operator/actions/runs/2909282076

I will look into figuring out how to handle it, but in the meanwhile, please do review this PR anyway, since the failure is not due to the changes here.

File: operatorconfig/moduleconfig/common/version-values.yaml [hunk 1]
```diff
@@ -7,3 +7,6 @@ powerscale:
   v2.3.0:
       authorization: "v1.3.0"
       replication: "v1.3.0"
+  v2.4.0:
+      authorization: "v1.4.0"
+      replication: "v1.4.0"
```
Comment (rensyct) [inline operatorconfig/moduleconfig/common/version-values.yaml:12]: Should replication point to v1.3.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: attacher
+          image: k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:138]: Update the sidecar version to v3.4.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: external-health-monitor
+          image: k8s.gcr.io/sig-storage/csi-external-health-monitor-controller:v0.4.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:155]: Update the sidecar version to v5.0.1

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: provisioner
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:175]: Update the sidecar version to v3.1.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: snapshotter
+          image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.1
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:197]: Update the version to v5.0.1

File: operatorconfig/driverconfig/powerscale/v2.4.0/node.yaml [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+              mountPath: /dev
+            - name: certs
+              mountPath: /certs
+              readOnly: true
+            - name: isilon-configs
+              mountPath: /isilon-configs
+            - name: csi-isilon-config-params
+              mountPath: /csi-isilon-config-params
+        - name: registrar
+          image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.4.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/node.yaml:141]: Update the version to v2.5.1

**Review by JacobGros — APPROVED** [review]

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+              matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - <DriverDefaultReleaseName>-controller
+            topologyKey: kubernetes.io/hostname
+
+      containers:
+        - name: resizer
+          image: k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:121]: Please change this version to v1.4.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: external-health-monitor
+          image: k8s.gcr.io/sig-storage/csi-external-health-monitor-controller:v0.5.1
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:155]: should be v0.5.0

**Review by JacobGros — APPROVED** [review]

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+              matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - <DriverDefaultReleaseName>-controller
+            topologyKey: kubernetes.io/hostname
+
+      containers:
+        - name: resizer
+          image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:121]: sidecar to be updated to v1.5.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: attacher
+          image: k8s.gcr.io/sig-storage/csi-attacher:v3.4.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:138]: sidecar needs to be updated v3.5.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: external-health-monitor
+          image: k8s.gcr.io/sig-storage/csi-external-health-monitor-controller:v0.5.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:155]: sidecar needs to be updated to v0.6.0

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: provisioner
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v3.1.0
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:175]: sidecar needs to be updated to v3.2.1

File: operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+            - "--leader-election-lease-duration=15s"
+            - "--leader-election-retry-period=5s"
+          env:
+            - name: ADDRESS
+              value: /var/run/csi/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/run/csi
+        - name: snapshotter
+          image: k8s.gcr.io/sig-storage/csi-snapshotter:v5.0.1
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerscale/v2.4.0/controller.yaml:197]: sidecar needs to be updated to v6.0.1

**Review by rensyct — CHANGES_REQUESTED** [review]

**Review by rensyct — APPROVED** [review]

**Review by JacobGros — APPROVED** [review]

**Pull request merged**